### PR TITLE
Set up association

### PR DESCRIPTION
**Pull request by leloufadel**
in this PR : 
 Created a class Classroom .
 Created the has-many/belongs-to relationship between Classroom and Student.
 Created the has-many side (a classroom has many students).
 Created the belongs-to side (a student belongs to a classroom).
 Ensured that when adding a student to a classroom it also sets the classroom for the student.
 Ensured that when setting the classroom for a student it also adds it to the classrooms' students.
 Created a class Book with @titleand @author instance variables
 Created a class Rental with @Date instance variable,
 Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.
 Created the has-many side of BookandRental`
 Created the belongs-to side of Rental and Book
 Created the has-many side of Person and Rental
 Created the belongs-to side of Rental and Person
 Modified the constructor of Rental so Book and Person
In addition, I had fulfilled the following default requirements:

Make sure that there are no linter errors.
Used correct GitHub flow.
Documented my work in a professional way.
Follow our list of best practices for Ruby.

### CODE REVIEW

**Review by rloterh — COMMENTED** [review]
Hi Lala, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Kindly ensure that your ReadMe is professionally documented; Let's add the _Future Features_ section of your ReadMe, and it should include 1-3 potential features to be added later. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -1,15 +1,21 @@
 require './nameable'
-
+require './rental'
 class Person < Nameable
   attr_accessor :name, :age
   attr_reader :id
```
Comment (roshan-bajgain) [inline person.rb]: - [ ] I would suggest you please use `Getters and Setters for @rentals` for example:
``attr_accessor :name, :age, :rental``
However, if you foresee a situation where you might need to access or modify @rental from outside the Person class in the future, you can add attr_accessor :rental to make it accessible externally. Please implement it

File: student.rb [hunk 2]
```diff
@@ -6,6 +6,11 @@ def initialize(classroom, id, age, name: 'Unknown', parent_permission: true)
     @classroom = classroom
   end
 
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.students.push(self) unless classroom.students.includes?(self)
```
Comment (roshan-bajgain) [inline student.rb:13]: - [ ]  Please as mentioned also provide ``attr_reader :classroom`` when you use attr_reader, it only creates a getter method So You can use classroom to retrieve the value of the @classroom instance variable.

File: classroom.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label, :students
+
```
Comment (roshan-bajgain) [inline classroom.rb]: - [ ] Please provide `attr_reader :students` when you use attr_reader, it only creates a getter method So You can use students to retrieve the value of the @Students instance variable.

**Review by roshan-bajgain — CHANGES_REQUESTED** [review]
Hi @leloufadel, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- No linter errors. ✔️
- Gitflow is followed. ✔️
- Good job on creating the association ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @leloufadel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- Gitflow is followed. ✔️
- Good job on implementing the changes 😃 ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**